src/js/actions/layers.js

```diff
@@ -1296,9 +1296,16 @@ define(function (require, exports) {
                 var allLayerIDs = collection.pluck(results, "ID")
                     .reduce(function (allLayerIDs, layerIDs) {
                         return allLayerIDs.concat(layerIDs);
-                    }, []);
+                    }, []),
+                    allLayers = allLayerIDs.map(function (currentVal) {
+                        return { id: currentVal }
+                    });
 
-                return this.transfer(addLayers, document, allLayerIDs, undefined, false);
+                return this.transfer(addLayers, document, allLayerIDs, undefined, false)
+                    .bind(this)
+                    .then( function () {
+                        return this.transfer(selectLayer, document,  Immutable.List(allLayers));
```

Comment:
best rebase first, since AFAICT this action has since changed names (now: 'select')